### PR TITLE
Editable reblogs manual posts

### LINKED ISSUE #507: Readmores converting to hr lines with editable reblogs enabled
[This post] (http://txmekeeper.tumblr.com/post/129317891468/txmekeeper-txmekeeper-test-post-keep) is an example of what I've been getting when I try to insert a readmore into a post with xinbox enabled. My extension configuration export is [here] (https://gist.github.com/anonymous/601c3893bd5ac1ddbef1), and there's a screenshot helpfully provided by @blackjackkent of the offending post on the dash.

Feel free to reblog that post as much as you want to test it, it's on an unused sideblog.

nolines

### LINKED ISSUE #523: Editable Reblogs fails when images inserted into posts
Whenever I go to input a gif, and then try to put text under it, the text goes away completely. Then when I edit it, the text is still gone.

### PR DESCRIPTION
**Pull request by blackjackkent**
Fixes #507 
Fixes #523 
...hopefully.

Please test with great vigor and intensity.

### CODE REVIEW

**Comment by dlmarquis** [comment]
I found the fix to people accidentally taking ownership over photo captions, etc. There's no need to check if the previous user made a comment, and the all_quotes_text=reblog_content around https://github.com/new-xkit/XKit/pull/605/files#diff-8af2f38a5ebca8d6d23e9b0108fb3b9aR83 is where comment ownership gets screwed up. We only need the code in the else statement.

I was going to make a PR, but I didn't want to just screw you over.

File: Extensions/editable_reblogs.js [hunk 1]
```diff
@@ -90,81 +105,156 @@ XKit.extensions.editable_reblogs = new Object({
...
-		if (reblog_tree.length <= 0) {
-			return;
+	build_common_svc_request: function() {
+		var request = {};
+		var location_path = window.location.pathname;
+		var location_items = location_path.split("/");
+		location_items.shift();
+		request.form_key = XKit.interface.form_key();
+		request.channel_id = $('.post-form--header .tumblelog-select .caption').text();
+		request.detached = true; //?
```
Comment (bvtsang) [inline Extensions/editable_reblogs.js]: Does the comment here mean that you don't know what this is for?

Comment (blackjackkent) [inline Extensions/editable_reblogs.js]: Yes. :P

**Comment by bvtsang** [comment]
If you don't mind, I'm going to be (continually) dumping my (regression) testing notes here.

For the Rich Text Editor and HTML editors:

- Basic functionality
  - [ ] Text posts
  - [ ] Photo posts
  - [ ] Links
  - [ ] Audio
  - [ ] Video
- Read mores
  - [ ] Read mores don't convert to `hr`s
  - [ ] ...for multiple read mores in the same post
- [ ] When inserting an image into a post and adding text beneath it, the text is preserved

For the Markdown editor, when the user clicks on the reblog button, a warning should come up telling them to switch editors.

**Comment by nightpool** [comment]
@bvtsang re: videos—could you repro that on pre-3 ER? I had been having some similar problems (although they were probably not related to ER, as they were quote posts and so didn't have reblog_trees)

**Comment by bvtsang** [comment]
@nightpool Pre-ER 3.0, I can see the video with a gray rectangle below it.

**Comment by nightpool** [comment]
just the version change I think

**Comment by nightpool** [comment]
also fixes #675 iirc.

File: Extensions/editable_reblogs.js [hunk 2]
```diff
@@ -85,81 +102,156 @@ XKit.extensions.editable_reblogs = new Object({
...
+		var new_button_content = "<div class='control right xkit-editable-reblogs-control'>"
+			+ '<div class="post-form--save-button" data-subview="savePostButton">'
+			+ '<button class="flat-button blue caption create_post_button xkit-editable-reblogs-button" id="xkit-editable-reblogs-post">Post</button>'
+			+ '<button class="flat-button blue caption create_post_button xkit-editable-reblogs-button" id="xkit-editable-reblogs-queue">Queue</button>'
+			+ '<button class="flat-button blue caption create_post_button xkit-editable-reblogs-button" id="xkit-editable-reblogs-draft">Draft</button>'
+			+ '</div></div>';
+		$('.post-form--controls .controls-container').append(new_button_content);
+		$('body').on('click', '#xkit-editable-reblogs-post', XKit.extensions.editable_reblogs.send_post_request);
+		$('body').on('click', '#xkit-editable-reblogs-queue', XKit.extensions.editable_reblogs.send_queue_request);
+		$('body').on('click', '#xkit-editable-reblogs-draft', XKit.extensions.editable_reblogs.send_draft_request);
```
Comment (bvtsang) [inline Extensions/editable_reblogs.js]: These need to be removed in the `destroy` function.

**Comment by nightpool** [comment]
where are we on this?

**Comment by bvtsang** [comment]
Awaiting the addition of cleanup in the `destroy` function.

**Comment by bvtsang** [comment]
LGTM

@homu r+

**Pull request merged**